components/mediation/event-sink/org.wso2.micro.integrator.event.sink/src/main/java/org/wso2/micro/integrator/event/sink/EventSink.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.event.sink;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.micro.core.util.CryptoException;
+import org.wso2.micro.core.util.CryptoUtil;
+import org.wso2.carbon.databridge.agent.DataPublisher;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointAgentConfigurationException;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointAuthenticationException;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointConfigurationException;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointException;
+import org.wso2.carbon.databridge.agent.util.DataEndpointConstants;
+import org.wso2.carbon.databridge.commons.exception.TransportException;
+
+import java.nio.charset.Charset;
+
+public class EventSink {
+    private static final Log log = LogFactory.getLog(EventSink.class);
+    private String name;
+    private String receiverUrlSet;
+    private String authenticationUrlSet;
+    private String username;
+    private String password;
+    private DataPublisher dataPublisher;
+
+    public EventSink() {
+    }
+
+    public EventSink(String name, String username, String password, String receiverUrlSet, String authenticationUrlSet) {
```

Comment:
Is this constructor used?

Reply:
Removed the previous constructor and refactored to use this constructor in bd10bd3